pipeline/frontend/yaml/compiler/compiler.go

```diff
@@ -82,6 +82,12 @@ func New(opts ...Option) *Compiler {
 func (c *Compiler) Compile(conf *yaml.Config) *backend.Config {
 	config := new(backend.Config)
 
+	if !conf.MatchConstraints(c.metadata) {
+		// This pipeline does not match the configured filter so return an empty config and stop further compilation.
```

Comment:
this is a nice catch ... but has one downside: on multi pipeline builds, if one pipeline depend on this, it will fail 
-> we have to determine if a pipeline with no steps failed or is successful and somehow represent this

Reply:
(can we move this in a own pull ...? )

Reply:
Sure thing. I'll grab that once back at my terminal. Its just efficiency though, so you can just remove it.

Reply:
Hum... @6543 I don't understand. I think this would work as expected. I am returning an empty config in the case where the pipeline dose not match therefore multiple pipelines should execute. The pipeline config will be filtered on no steps later on.

I am relying on the check of empty number of steps so it would run.

E.g. these two pipelines work as expected (tested), only on_push runs.

1 on_release.yml
```yaml
when:
  - event:
      - pull_request
      - release

pipeline:
  envs:
    image: ubuntu:latest
    commands:
      - env
```

2 on_push.yml
```yaml
# Will only run afer first, and in the case of a pull request or release.
when:
  - event:
      - push

pipeline:
  always_run_on_push:
    image: ubuntu:latest
    commands:
      - echo "Runs on push"
```